### PR TITLE
Revert fix for GEODE-8231 

### PR DESCRIPTION
**Pull request by pdxcodemonkey**
"remove bucket server location from metadata when …server down (#615)"

This reverts commit 85951d7e23a9cb493fde16d09a56c785d09d6ba5.

### CODE REVIEW

File: cppcache/src/ClientMetadata.cpp [hunk 1]
```diff
@@ -114,24 +114,14 @@ const std::string& ClientMetadata::getColocatedWith() {
...
-        break;
-      }
-    }
-  }
-}
-
 void ClientMetadata::getServerLocation(
     int bucketId, bool tryPrimary,
     std::shared_ptr<BucketServerLocation>& serverLocation, int8_t& version) {
+  // ReadGuard guard (m_readWriteLock);
```
Comment (echobravopapa) [inline cppcache/src/ClientMetadata.cpp:120]: can we lose this dead code while we're in here... thx

File: cppcache/src/ClientMetadata.cpp [hunk 2]
```diff
@@ -153,15 +143,30 @@ void ClientMetadata::getServerLocation(
...
-    LOGFINER("returning random & m_bucketServerLocationsList size is: %zu",
-             m_bucketServerLocationsList.size());
     RandGen randgen;
     serverLocation = m_bucketServerLocationsList[bucketId].at(randgen(
         static_cast<int>(m_bucketServerLocationsList[bucketId].size())));
   }
   // return m_bucketServerLocationsList[bucketId].at(0);
 }
 
+/*
```
Comment (echobravopapa) [inline cppcache/src/ClientMetadata.cpp:153]: here too

**Review by echobravopapa — DISMISSED** [review]

File: cppcache/src/ClientMetadata.cpp (hunk 2 above)
Comment (echobravopapa) [reply cppcache/src/ClientMetadata.cpp:153]: disregard, we'll clean up post revert...

**Review by echobravopapa — APPROVED** [review]

**Comment by alb3rtobr** [comment]
This PR also reverts code from GEODE-8565, which was an addition over GEODE-8231 solution.
So I think this PR should contain the revert of GEODE-8688, GEODE-8565 and GEODE-8231.

**Comment by pdxcodemonkey** [comment]
> This PR also reverts code from GEODE-8565, which was an addition over GEODE-8231 solution.
> So I think this PR should contain the revert of GEODE-8688, GEODE-8565 and GEODE-8231.

Agree, this is the revert of all 3 of these.  This is just a title change for the PR, though, right?  I have reviewed the 3 earlier PRs and see no remaining code from them...

**Review by mmartell — APPROVED** [review]

**Comment by albertogpz** [comment]
I wonder if it would make more sense to write a JIRA to fix the problem caused by the GEODE-8231 PR instead of reverting 3 PRs that have been for months in the develop branch.

I do not know if I understood correctly the issue but wouldn't a possible solution be to change the eventId of the "leftover" keys (the keys for those puts that were not sent because the server went down) before sending them in the new thread to another server?

If we did so, we could keep the code in the "PRs to be reverted", fix the issue and have a test case to verify that everything works as expected.

**Comment by pdxcodemonkey** [comment]
> I wonder if it would make more sense to write a JIRA to fix the problem caused by the GEODE-8231 PR instead of reverting 3 PRs that have been for months in the develop branch.
> 
> I do not know if I understood correctly the issue but wouldn't a possible solution be to change the eventId of the "leftover" keys (the keys for those puts that were not sent because the server went down) before sending them in the new thread to another server?
> 
> If we did so, we could keep the code in the "PRs to be reverted", fix the issue and have a test case to verify that everything works as expected.

I'm looking into it - don't think this one is going to be merged right away, because the fix may be simpler and IMO lower-risk than reverting all this code.  Writing the test is going to be tough - we had to modify Geode server code in order to reproduce the issue in the first place.  At any rate I'll update things here.

**Comment by albertogpz** [comment]
> > I wonder if it would make more sense to write a JIRA to fix the problem caused by the GEODE-8231 PR instead of reverting 3 PRs that have been for months in the develop branch.
> > I do not know if I understood correctly the issue but wouldn't a possible solution be to change the eventId of the "leftover" keys (the keys for those puts that were not sent because the server went down) before sending them in the new thread to another server?
> > If we did so, we could keep the code in the "PRs to be reverted", fix the issue and have a test case to verify that everything works as expected.
> 
> I'm looking into it - don't think this one is going to be merged right away, because the fix may be simpler and IMO lower-risk than reverting all this code. Writing the test is going to be tough - we had to modify Geode server code in order to reproduce the issue in the first place. At any rate I'll update things here.

Great. Thanks

**Comment by echobravopapa** [comment]
@albertogpz this change alters consistent behavior between clients (java & native) along with data loss, in the near term the most prudent approach is to remove the code causing the issues.

**Comment by pivotal-jbarrett** [comment]
> @albertogpz this change alters consistent behavior between clients (java & native) along with data loss, in the near term the most prudent approach is to remove the code causing the issues.

@echobravopapa I have to say I disagree. The change to fix the PUTALL problem is simple. The impact the targeted removal of bucket metadata is more positive since it affects all single value operations. The only operations degraded are the multikey operations, which are more rare. The timing to get either change in is the same so I would vote to get the more complete fix in.

**Comment by pdxcodemonkey** [comment]
Closed because this a) doesn't really solve the problem, and b) is much more invasive than the actual fix (see GEODE-9147).

**Pull request closed without merging**